c++/src/kj/mutex.c++

```diff
@@ -263,6 +263,13 @@ bool Mutex::lock(Exclusivity exclusivity, Maybe<Duration> timeout, LockSourceLoc
 #endif
       break;
     case SHARED: {
+#if KJ_CONTENTION_WARNING_THRESHOLD
+      // We could have the exclusive mutex tell us how long it was holding the lock. That would be
+      // the nicest. However, I'm hesitant to bloat the structure. I suspect having a reader tell
+      // us how long it was waiting for is probably a good proxy.
+      auto contentionWaitStart = kj::systemPreciseMonotonicClock().now();
```

Comment:
Can you move this down so that in the common, uncontended case, we don't read the clock at all?

Reply:
It complicates the logic slightly but seems reasonable. Let me know what you think.